### ISSUE TITLE
createcachetable does not seem to work

### ISSUE THREAD

**Issue by gitsilvan** [bernardopires/django-tenant-schemas#348]
Nothing seems to happen when I call `python manage.py createcachetable`. Calling it with `--dry-run` doesn't output anything either. I want to use db caching. Settings as follows:

```
CACHES = {
    'default': {
        'BACKEND': 'django.core.cache.backends.db.DatabaseCache',
        'LOCATION': 'cache_table',
        'KEY_FUNCTION': 'tenant_schemas.cache.make_key',
    }
}
```

The db is working besided that.

**Comment by gitsilvan** [comment]
After removing this line from `createcachetable.py` the command works as expected:
```
if not router.allow_migrate_model(database, cache.cache_model_class):
            return
```
So apparently migrating the cache table is not allowed.

**Comment by AlexvEck** [comment]
This seems unrelated to django-tenant-schemas as only providing the  KEY_FUNCTION should not affect the normal functioning of caching. Have you been able to figure out more about why this happens?

**Comment by gitsilvan** [comment]
No, sorry – since it did work with the hackish workaround I posted I didn't have to look more into it…

**Comment by durkode** [comment]
I also had this issue, and worked out what the issue is. in the tenant_schemas.routers.TenantSyncRouter class, allow_migrate method:

       if connection.schema_name == get_public_schema_name():
            if app_label not in app_labels(settings.SHARED_APPS):
                return False
        else:
            if app_label not in app_labels(settings.TENANT_APPS):
                return False

The router assumes that every model that is passed through allow_migrate will have the app_label installed in either SHARED_APPS or TENANT_APPS, however using the database as a cache doesn't fall into either of these cases.

I'm not sure what the fix is here. For a workaround I did what @gitsilvan suggested and created my own management command that removed the router check. Not sure what the correct fix is for django-tenant-schemas as I am unsure of what other edge cases there are for model migrations that aren't in installed apps.
